portal/common/lib/resource.ts

```diff
@@ -41,38 +47,25 @@ export async function fetchResource(
         return HttpStatusCodes.TOO_MANY_REDIRECTS;
     }
 
-    // Initiate a pre-fetch for the checkRedirect operation without resolving it.
-    // We don't need the result yet, but it's useful if we will need
-    // it later, so we don't have to loose time.
-    const checkRedirectPromise = checkRedirect(client, objectId);
+    let redirectId  = await checkRedirect(client, objectId);
+    if (redirectId) {
+        fetchResource(client, redirectId, path, seenResources, depth + 1)
+    }
```

Comment:
Question/important: i think we still want to do the background prefetch of the redirect, so that we first check if there is a resource, and only if there isn't we redirect. Why the change?